app/controllers/SessionTimeoutPageController.scala

```diff
@@ -0,0 +1,141 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.{Inject, Singleton}
+import java.time.Instant
+
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request, Result, Session}
+import config.{AppConfig, ErrorHandler}
+import models.errors._
+import models.ui.Text
+import services.GuidanceService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.http.logging.SessionId
+import uk.gov.hmrc.http.SessionKeys._
+import views.html.session_timeout
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class SessionTimeoutPageController @Inject()(
+                                              appConfig: AppConfig,
+                                              service: GuidanceService,
+                                              errorHandler: ErrorHandler,
+                                              mcc: MessagesControllerComponents,
+                                              view: session_timeout)
+    extends FrontendController(mcc)
+    with I18nSupport {
+
+    val logger = Logger(getClass)
+
+    def getPage(processCode: String): Action[AnyContent] = Action.async { implicit request =>
+
+      implicit val messages: Messages = mcc.messagesApi.preferred(request)
+
+      val sessionId: Option[SessionId] = hc.sessionId
+
+      sessionId match {
+        case Some(id) => {
+          if(hasSessionExpired(request.session)) {
+            Future.successful(createYourSessionHasExpiredResponse(messages("session.timeout.header.title"), processCode))
+          } else {
+            manageDeleteYourAnswersRequest(id, processCode)
+          }
+        }
+        case None => Future.successful(createYourSessionHasExpiredResponse(messages("session.timeout.header.title"), processCode))
+      }
+    }
+
+  def manageDeleteYourAnswersRequest(sessionId: SessionId, processCode: String)(implicit request: Request[_]): Future[Result] = {
+
+    // Refresh cache to remove answers
+    implicit val messages: Messages = mcc.messagesApi.preferred(request)
+
+    service.getProcessContext(sessionId.value).flatMap {
+      case Right(processContext) if (processCode != processContext.process.meta.processCode) => {
+        logger.error(s"Unexpected process code encountered when removing session after timeout warning. Expected code $processCode; actual code ${processContext.process.meta.processCode}")
+        Future.successful(InternalServerError(errorHandler.internalServerErrorTemplate))
+      }
+      case Right(processContext) => {
+        service.removeSession(sessionId.value).map {
```

Comment:
This isnt necessary as mongo will delete the session repo document automatically, what is important is to remove the current sessionId from the session

Reply:
removed